cascading-core/src/main/java/cascading/tuple/TupleEntrySchemeIterator.java

```diff
@@ -165,6 +165,7 @@ private TupleEntry getNext() throws IOException
     if( !hasWaiting && inputIterator.hasNext() )
       {
       sourceCall.setInput( wrapInput( inputIterator.next() ) );
+      scheme.sourcePrepare( flowProcess, sourceCall );
```

Comment:
is there going to be a big hit to call this each time? Is this the right fix vs in elephantbird?

Reply:
this will hit only once we switch between recordReaders. 

Unfortunately, it's impossible to fix in every possible recordRead. many of them written in the same way which used in ElephantBird. I think it's a problem with "elegant" design of recordReader in hadoop, `createKey` and `createValue` creates mutable objects which passed to method `next` which supposed to mutate them.